### DISCUSSION TITLE
getting compute error when i try to return an experiment type in a nada program

### DISCUSSION THREAD

**Discussion by Thompsonmina** [NillionNetwork/discussions#33] [Q&A] [answered]
The docs and sample code show nada code that returns complex types, but I have observed that whenever I try to return a complex type in my code, I get a compute error. 

Are complex return types not yet supported?

Examples
`py_nillion_client.ComputeError: reading computation output: Array { inner_type: Tuple { left_type: SecretInteger, right_type: SecretInteger }, size: 5 } is not a primitive type`

`py_nillion_client.ComputeError: reading computation output: Tuple { left_type: Array { inner_type: SecretInteger, size: 5 }, right_type: Array { inner_type: SecretInteger, size: 5 } } is not a primitive type`

**Comment by navasvarela** [comment, marked as answer]
Hi, thank you for reporting this issue @Thompsonmina!

I can see in the error message that you are using the Python client. There is currently a limitation that we don't allow complex types to be returned by our JS or Python clients. We are aiming at fixing this for the next release. In the meanwhile, if you give me more details about what you are trying to accomplished we might be able to suggest a workaround or alternative path.

**Reply by Thompsonmina** [reply]
hey, I am working on a program where it would have been more convenient to return an array but strictly speaking I could probably work around it. I just wanted to confirm it wasn't something on my end that was the issue. Thanks for the response.

**Reply by navasvarela** [reply]
You are welcome!
